matchengine/engine.py

```diff
@@ -220,103 +229,227 @@ def create_trial_tree(self, raw_data, no_validate=False):
-                        'sample_id': sample_id,
+                    # add genomic information and alterations that matched per sample id
+                    genomic_info = {
                         'match_type': is_variant,
                         'genomic_alteration': alteration
-                    } for sample_id in matched_sample_ids]
+                    }
 
-                else:
-                    for item in results:
+                    # copy genomic document projection into match
+                    for field in proj:
+                        if field in item:
+                            if field == '_id':
+                                genomic_info['genomic_id'] = item[field]
+                            else:
+                                genomic_info[field.lower()] = item[field]
 
-                        # format the genomic alteration that matched
-                        alteration, is_variant = format_genomic_alteration(item, g)
+                    # add unique matches by sample id
+                    matched_genomic_info.append(genomic_info)
 
-                        # add genomic information and alterations that matched per sample id
-                        genomic_info = {
-                            'match_type': is_variant,
-                            'genomic_alteration': alteration
-                        }
+                matched_sample_ids = set(item['SAMPLE_ID'] for item in results)
 
-                        # copy genomic document projection into match
-                        for field in proj:
-                            if field in item:
-                                if field == '_id':
-                                    genomic_info['genomic_id'] = item[field]
-                                else:
-                                    genomic_info[field.lower()] = item[field]
+        return matched_sample_ids, matched_genomic_info
+
+    def oncokb_match(self, conditions):
+        """
+        Runs genomic query based on 'annotated_variant' against Mongo database and returns a set of sample ids that matched
+        :param conditions: query conditions, value of genomic node with the trial match tree
+        :param db: database connection
+        :returns
+            matched_sample_ids: set of matched sample ids
+            matched_genomic_info: genomic information regarding each match
+        """
+        results = list()
+        matched_genomic_info = []
+        hugo_symbol = conditions['hugo_symbol']
+        annotated_variant = conditions['annotated_variant']
+        negative_query = False
+
+        # Negative queries will be run as positive queries and the matched sample ids will be subtracted from
+        # the set of all sample ids in the database
+        if hugo_symbol.startswith('!'):
+            negative_query = True
+            # remove "!"
+            hugo_symbol = hugo_symbol[1:]
+
+        if annotated_variant.startswith('!'):
```

Comment:
not logic makes the whole matching to false according to the following statement. What happens when hugo: `BRAF`, annotated_variant: `!V600E`?